app/main.py

```diff
@@ -2,5 +2,19 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def inner(*args) -> str:
+        if func.__name__ not in cached_results:
+            print("Calculating new result")
+            cached_results[func.__name__] = {args: func(*args)}
+        else:
+            if args in cached_results[func.__name__]:
+                print("Getting from cache")
+            else:
+                print("Calculating new result")
+                cached_results[func.__name__][args] = func(*args)
```

Comment:
This is a code efficiency issue. You are calling `func` multiple times. This can be inefficient if `func` is a computationally expensive function. Store the result in a variable and reuse it. For example:

python
result = func(*args)
cached_results[func.__name__][args] = result
```

This way, you only call `func` once and reuse the result.